stan/math/opencl/matrix_cl.hpp

```diff
@@ -46,11 +48,84 @@ class matrix_cl {
   int cols() const { return cols_; }
 
   int size() const { return rows_ * cols_; }
-
   const cl::Buffer& buffer() const { return oclBuffer_; }
-
   matrix_cl() : rows_(0), cols_(0) {}
 
+  matrix_cl(cl::Buffer& A, const int& R, const int& C) : rows_(R), cols_(C) {
```

Comment:
Why are R and C `const` references? Is there any benefit of them not being just `const int`?

Reply:
tbh habit, these are simple types so idt there's a big benefit of doing them by ref

Reply:
Would you mind if I change this to `const int`s? I think we pass these stuff by value in other places?